### PR TITLE
Add nagger docker build/push job

### PR DESCRIPTION
**Pull request by tomaszklak**

### CODE REVIEW

File: ci/docker/release/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
         # libdrop deps
         cmake \
         # Libtelio dependencies
-        quilt; \
-    apt-get clean
+        quilt && \
+    apt-get clean && \
+    cargo install --version 2.4.3 cargo-dylint dylint-link && \
+    # This is the last 1.72.0 nightly build that works with rust-clippy.
+    rustup update nightly-2023-07-02
```
Comment (LukasPukenis) [inline ci/docker/release/Dockerfile:22]: Is there a reason to use nightly build instead of stable one?

File: ci/build_image.sh [hunk 2]
```diff
@@ -1,42 +1,30 @@
 #!/usr/bin/env bash
 set -euxo pipefail
 
-source $(dirname $0)/env.sh
-
```
Comment (LukasPukenis) [inline ci/build_image.sh:5]: env.sh isnt present in this repository, was it always erroring then? Just curious, nice to see external dependency removed

File: ci/build_image.sh [hunk 3]
```diff
@@ -1,42 +1,30 @@
...
-    *)
-        echo "Unrecognized image '${2:-}'"
-        print_help
-esac
+SCRIPT_DIR="$PWD/ci"
+BASE_DIR=$PWD
+# Github pull request has github.ref_name (which is used to set this) in
+# format "${pull_request_id}/merge" and '/' is not a valid docker tag so
+# we convert it to '_'.
+CI_COMMIT_TAG=$(echo $CI_COMMIT_TAG | tr '/' '_')
```
Comment (LukasPukenis) [inline ci/build_image.sh:14]: Nitpick: I think the name is misleading as it's the image_tag name, could be named as such instead

File: ci/build_image.sh [hunk 4]
```diff
@@ -1,42 +1,30 @@
...
-        DOCKERFILE="$SCRIPT_DIR/docker/release/Dockerfile"
-        IMAGE_TAG="${CONTAINER_REGISTRY}/low-level-hacks/automation/nagger/release:${CI_COMMIT_TAG}"
-        ARGS="--build-arg CONTAINER_REGISTRY=${CONTAINER_REGISTRY} \
-              --build-arg NAGGER_BUILD_TAG=${NAGGER_BUILD_TAG}"
-        ;;
-    *)
-        echo "Unrecognized image '${2:-}'"
-        print_help
-esac
+SCRIPT_DIR="$PWD/ci"
```
Comment (Hasan6979) [inline ci/build_image.sh:9]: nit : SCRIPT_DIR="BASE_DIR/ci" instead ?

File: ci/build_image.sh [hunk 5]
```diff
@@ -1,42 +1,30 @@
 #!/usr/bin/env bash
 set -euxo pipefail
 
-source $(dirname $0)/env.sh
-
 print_help() {
     echo "Usage: ci/build_image.sh <build|push> <build|release>"
```
Comment (LukasPukenis) [inline ci/build_image.sh:5]: I assume this needs an update as the second parameter is now unavailable

File: ci/build_image.sh [hunk 6]
```diff
@@ -1,42 +1,30 @@
...
+DOCKERFILE="$SCRIPT_DIR/docker/release/Dockerfile"
+IMAGE_TAG="${CONTAINER_REGISTRY}/nordsecurity/nagger:${CI_COMMIT_TAG}"
+ARGS="--build-arg CONTAINER_REGISTRY=${CONTAINER_REGISTRY}"
 
 case "${1:-}" in
     build)
         docker build -f "$DOCKERFILE" -t "$IMAGE_TAG" ${BASE_DIR} ${ARGS}
         ;;
     push)
-        if $(docker manifest inspect $IMAGE_TAG &> /dev/null); then
```
Comment (Hasan6979) [inline ci/build_image.sh:33]: Why is this removed ?

File: ci/build_image.sh [hunk 7]
```diff
@@ -1,42 +1,30 @@
 #!/usr/bin/env bash
```
Comment (LukasPukenis) [inline ci/build_image.sh:1]: Nitpick: build_image tells me(the reader) that it's all about building the image so it looks a bit funny to have `build_image.sh build`. Maybe renaming the file to `image.sh` would make it neater? `image.sh build` and then `image.sh push`

**Review by Hasan6979 — COMMENTED** [review]
+ 1.0

File: ci/install_docker.sh [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+chmod a+r /etc/apt/keyrings/docker.gpg
+
+# Add the repository to Apt sources:
+echo \
+  "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+  "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+  tee /etc/apt/sources.list.d/docker.list > /dev/null
+apt-get update
+
+apt-get --yes install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
```
Comment (LukasPukenis) [inline ci/install_docker.sh:18]: Maybe we can use already pre-made action for this such as: https://github.com/marketplace/actions/setup-docker ?

File: ci/build_image.sh (hunk 6 above)
Comment (tomaszklak) [reply ci/build_image.sh:33]: This is removed to make it possible to push updated image version for new commits on the branch (so that the author of the branch can use that image for testing). I can imagine that this guard to not override images was useful when we worked with the images manually but since we move all the work to automated gh jobs there is no reason to worry that a released tag will be overwritten manually by accident since there is no reason to do anything like that locally.

File: ci/docker/release/Dockerfile (hunk 1 above)
Comment (tomaszklak) [reply ci/docker/release/Dockerfile:22]: dylint requires nightly toolchain.

File: ci/build_image.sh (hunk 2 above)
Comment (tomaszklak) [reply ci/build_image.sh:5]: The old gitlab version has the `env.sh` file.

File: ci/install_docker.sh (hunk 8 above)
Comment (tomaszklak) [reply ci/install_docker.sh:18]: That action hardcodes the usage of sudo (which is not present in the `rust:1.72` image). So to use it I would have to add sudo installation and setup which I don't think is worth it. I think I would prefer to keep the official way of installing docker as is (see https://docs.docker.com/engine/install/debian/#install-using-the-repository).

File: ci/build_image.sh (hunk 4 above)
Comment (tomaszklak) [reply ci/build_image.sh]: Done :)

File: ci/build_image.sh (hunk 3 above)
Comment (tomaszklak) [reply ci/build_image.sh]: Done :)

File: ci/build_image.sh (hunk 5 above)
Comment (tomaszklak) [reply ci/build_image.sh]: Done :)

File: ci/build_image.sh (hunk 7 above)
Comment (tomaszklak) [reply ci/build_image.sh]: Done :)

**Comment by LukasPukenis** [comment]
+1

**Review by LukasPukenis — APPROVED** [review]
